cmake/modules/python.cmake

```diff
@@ -11,37 +11,7 @@ if (WIN32)
   set(ENV{PYTHONIOENCODING} "utf-8")
 endif()
 
-set(PYTHON_MINIMUM_REQUIRED 3.8)
-
-find_package(Deprecated COMPONENTS PYTHON_PREFER)
-
-if(NOT DEFINED Python3_EXECUTABLE AND DEFINED WEST_PYTHON)
-  set(Python3_EXECUTABLE "${WEST_PYTHON}")
-endif()
-
-if(NOT Python3_EXECUTABLE)
-  # We are using foreach here, instead of
-  # find_program(PYTHON_EXECUTABLE_SYSTEM_DEFAULT "python" "python3")
-  # cause just using find_program directly could result in a python2.7 as python,
-  # and not finding a valid python3.
-  foreach(candidate "python" "python3")
-    find_program(Python3_EXECUTABLE ${candidate})
-    if(Python3_EXECUTABLE)
-        execute_process (COMMAND "${Python3_EXECUTABLE}" -c
-                                 "import sys; sys.stdout.write('.'.join([str(x) for x in sys.version_info[:2]]))"
-                         RESULT_VARIABLE result
-                         OUTPUT_VARIABLE version
-                         ERROR_QUIET
-                         OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-       if(version VERSION_LESS PYTHON_MINIMUM_REQUIRED)
-         set(Python3_EXECUTABLE "Python3_EXECUTABLE-NOTFOUND" CACHE INTERNAL "Path to a program")
-       endif()
-    endif()
-  endforeach()
-endif()
-
-find_package(Python3 ${PYTHON_MINIMUM_REQUIRED} REQUIRED)
+find_package(Python 3.8...3.12 COMPONENTS Interpreter)
```

Comment:
as minimum, please verify that this is not reintroduced: https://github.com/zephyrproject-rtos/zephyr/issues/24340

as well as verifying the different scenarios described in this PR are correctly handled: https://github.com/zephyrproject-rtos/zephyr/pull/24364

Just because this PR works in your environment doesn;t mean it work across  various environments, and we need to be a bit careful here.

But would be wonderful if later CMake version have improved in this regard.

Reply:
just verified myself, with CMake 3.26, then the issue reported in #24340 still exists.

I have Python 3.8 in path but have both 3.8 and 3.11 installed.
Zephyr picks up 3.8 (as expected because it meets the requirements) but TF-M (using vanilla FindPython3) picks up 3.11, which may lead to other surprises.
See also: https://github.com/zephyrproject-rtos/zephyr/pull/66219
and: https://discord.com/channels/720317445772017664/733037890514321419/1181702822245437510